src/Utils/ChangePosition.tsx

```diff
@@ -0,0 +1,31 @@
+import { SortableObject } from "../types/CommonTypes";
+
+export function changePosition<T extends SortableObject>(
+    listId: number, 
+    offset: number, 
+    collectionOfObjects: T[], 
+    setCollectionOfObjects: React.Dispatch<React.SetStateAction<T[]>>) :void {
+    const index = collectionOfObjects.findIndex((item) => item.id === listId);
+
+    if (index === -1) {
+        return;
+    }
+
+    const updatedList = [...collectionOfObjects];
```

Comment:
rewrite from 14 to 30th using .reduce()

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce